### PR TITLE
remove SecretGenerators from Kustomization.yaml

### LINKED ISSUE #692: remove SecretGenerator due to security concern
Need to address the security concerns of Kustomize in order to integrate it into Kubectl 
https://github.com/kubernetes/enhancements/pull/684

SecretGenerator is one concern since it can trigger arbitrary command that users put in.
The solution is to remove it from Kustomize.

### PR DESCRIPTION
**Pull request by Liujingfang1**
Fix #692 

remove `SecretGenerator` from kustomization.yaml
update unit tests

### CODE REVIEW

File: pkg/types/kustomization.go [hunk 1]
```diff
@@ -115,13 +115,6 @@ type Kustomization struct {
 	// the map will have a suffix hash generated from its contents.
 	ConfigMapGenerator []ConfigMapArgs `json:"configMapGenerator,omitempty" yaml:"configMapGenerator,omitempty"`
 
-	// SecretGenerator is a list of secrets to generate from
-	// local data (one secret per list item).
-	// The resulting resource is a normal operand, subject to
-	// name prefixing, patching, etc.  By default, the name of
-	// the map will have a suffix hash generated from its contents.
-	SecretGenerator []SecretArgs `json:"secretGenerator,omitempty" yaml:"secretGenerator,omitempty"`
```
[the remark this answers is not in the document]
Comment (Liujingfang1) [reply pkg/types/kustomization.go]: We should drop them. I'll update the change as well as dropping secretGenerator from the docs.

[the remark this answers is not in the document]
Comment (Liujingfang1) [reply pkg/types/kustomization.go]: Updated doc and removed `secretArgs`.

File: docs/kustomization.yaml [hunk 2]
```diff
@@ -83,40 +83,7 @@ configMapGenerator:
...
-- name: env_file_secret
-  # envCommand is similar to command but outputs lines of key=val pairs
-  # i.e. a Docker .env file or a .ini file.
-  # you can only specify one envCommand per secret.
-  envCommand: printf \"DB_USERNAME=admin\nDB_PASSWORD=somepw\"
-  type: Opaque
-
-# generatorOptions modify behavior of all ConfigMap and Secret generators
+# generatorOptions modify behavior of all ConfigMap generators
 generatorOptions:
```
Comment (liggitt) [inline docs/kustomization.yaml]: drop shell/timeoutSeconds examples?

Comment (Liujingfang1) [reply docs/kustomization.yaml]: done.

**Comment by monopole** [comment]
There's some pushback at #692, and that's just folks actively monitoring issues.

@liggitt Do you have any security concerns about keeping secret generator but making it work like configmap generator?

**Comment by liggitt** [comment]
> @liggitt Do you have any security concerns about keeping secret generator but making it work like configmap generator?

no, that seems reasonable as well

**Comment by Liujingfang1** [comment]
@monopole Are we going to replace `commands` by `files` or `literals` as in configmap generator?

**Comment by Liujingfang1** [comment]
Close this PR since we chose a different approach as in #703

**Pull request closed without merging**